src/llvm-alloc-helpers.cpp

```diff
@@ -245,6 +245,10 @@ void jl_alloc::runEscapeAnalysis(llvm::CallInst *I, EscapeAnalysisRequiredArgs r
                 required.use_info.addrescaped = true;
                 return true;
             }
+            if (required.pass.gc_loaded_func == callee) {
+                required.use_info.addrescaped = true;
```

Comment:
Seems slightly stricter than necessary, as we don't actually leak the address, we just have essentially a `haspreserve` and `hasload` attached to it